### PR TITLE
Add more tests for top_hits aggregation

### PR DESCRIPTION
**Pull request by nik9000**
Add unit tests for `TopHitsAggregator` and convert some snippets in
docs for `top_hits` aggregation to `// CONSOLE`.

Relates to #22278
Relates to #18160

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/aggregations/metrics/tophits/TopHitsAggregator.java [hunk 1]
```diff
@@ -122,7 +122,13 @@ public void collect(int docId, long bucket) throws IOException {
                     // In the QueryPhase we don't need this protection, because it is build into the IndexSearcher,
                     // but here we create collectors ourselves and we need prevent OOM because of crazy an offset and size.
                     topN = Math.min(topN, subSearchContext.searcher().getIndexReader().maxDoc());
-                    TopDocsCollector<?> topLevelCollector = sort != null ? TopFieldCollector.create(sort.sort, topN, true, subSearchContext.trackScores(), subSearchContext.trackScores()) : TopScoreDocCollector.create(topN);
+                    TopDocsCollector<?> topLevelCollector;
```
Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/search/aggregations/metrics/tophits/TopHitsAggregator.java]: 👍 much better readable

**Review by martijnvg — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/search/aggregations/metrics/tophits/TopHitsAggregator.java (hunk 1 above)
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/search/aggregations/metrics/tophits/TopHitsAggregator.java]: Thanks! I think it is pretty rare a ternary that spans multiple lines is more readable than a multi-line if.

**Comment by nik9000** [comment]
Thanks for reviewing @martijnvg!

master: d704a880e72579f89c7f88392623ad285e6a0d71
5.x:

**Pull request merged**